src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -1311,7 +1311,12 @@ public virtual bool TryValidateModel([NotNull] object model, string prefix)
                 modelState: ModelState,
                 modelExplorer: modelExplorer);
 
-            ObjectValidator.Validate(validationContext);
+            ObjectValidator.Validate(
+                validationContext,
+                new ModelValidationNode(modelName, modelExplorer.Metadata, model)
+                {
+                    BuildChildNodesUsingModel = true
+                });
```

Comment:
@rynowak the `ModelExplorer` instance is used just a couple of lines above when creating the `ModelValidationContext`.  The validation system is all about `ModelExplorer` right now.